packages/nimble-components/src/table-column/base/models/column-internals.ts

```diff
@@ -31,13 +31,18 @@ export interface ColumnInternalsOptions<
      * The tag to use to render the group header content for a column.
      * The element this tag refers to must derive from TableGroupHeaderView.
      */
-    readonly groupHeaderViewTag: string;
+    readonly groupHeaderViewTag?: string;
 
     /**
      * The names of events that should be delegated from the cell view to the column.
      */
     readonly delegatedEvents: readonly string[];
 
+    /**
+     * The names of slots that need to be forwarded into a cell.
+     */
+    readonly slotNames?: readonly string[];
```

Comment:
since this gets initialized in the constructor and is concretely always an array I think we can leave off the optional check

```suggestion
    readonly slotNames: readonly string[];
```

Reply:
This is the property on `ColumnInternalsOptions`, not `ColumnInternals`, so it doesn't get initialized in the constructor. Making it optional means that columns that don't care about slots don't need to explicitly set it to be an empty array.

Reply:
oops, my bad, makes sense